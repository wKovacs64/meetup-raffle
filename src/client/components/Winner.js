import React from 'react';
import PropTypes from 'prop-types';
import { css, cx } from 'emotion';

const Winner = ({ winner }) => {
  const bgImg = css`
    background-image: url(${winner.photoURL}), url('/user-placeholder.svg');
  `;

  return (
    <a
      className="flex flex-column dim link ba b--dark-blue dark-blue mb3 w4 shadow-5"
      href={winner.profileURL}
      target="_blank"
      rel="noopener noreferrer"
    >
      <span className={cx('db bg-center cover h4', bgImg)} role="img" />
      <div className="flex flex-grow-1 justify-center items-center bg-white">
        <span className="tc pa3" data-testid="name">
          {winner.name}
        </span>
      </div>
    </a>
  );
};

Winner.propTypes = {
  winner: PropTypes.shape({
    name: PropTypes.string.isRequired,
    photoURL: PropTypes.string.isRequired,
    profileURL: PropTypes.string.isRequired,
  }).isRequired,
};

export default Winner;
