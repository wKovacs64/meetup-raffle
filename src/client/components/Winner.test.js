import React from 'react';
import { render } from 'react-testing-library';
import Winner from './Winner';

const winner = {
  name: 'Pickle Rick',
  photoURL: 'https://i.imgur.com/3VhMoBD.png',
  profileURL: 'https://en.wikipedia.org/wiki/Pickle_Rick',
};

describe('Winner', () => {
  const { container, getByTestId } = render(<Winner winner={winner} />);

  it('renders', () => {
    expect(container).toMatchSnapshot();
  });

  it("includes the winner's name", () => {
    expect(getByTestId('name').textContent).toBe(winner.name);
  });
});
