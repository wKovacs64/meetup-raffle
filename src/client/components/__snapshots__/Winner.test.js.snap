// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Winner renders 1`] = `
<div>
  <a
    class="flex flex-column dim link ba b--dark-blue dark-blue mb3 w4 shadow-5"
    href="https://en.wikipedia.org/wiki/Pickle_Rick"
    target="_blank"
    rel="noopener noreferrer"
  >
    <span
      class="db bg-center cover h4 css-ho208l"
      role="img"
    />
    <div
      class="flex flex-grow-1 justify-center items-center bg-white"
    >
      <span
        class="tc pa3"
        data-testid="name"
      >
        Pickle Rick
      </span>
    </div>
  </a>
</div>
`;
